scripts/autotools/libmongoc/CheckCompression.m4

```diff
@@ -1,72 +1,111 @@
+PHP_ARG_WITH([mongodb-snappy],
+             [whether to enable Snappy for compression],
+             [AS_HELP_STRING([--with-mongodb-snappy=@<:@auto/yes/no@:>@],
+                             [MongoDB: Enable Snappy for compression [default=auto]])],
+             [auto],
+             [no])
+
+PHP_ARG_WITH([mongodb-zlib],
+             [whether to enable zlib for compression],
+             [AS_HELP_STRING([--with-mongodb-zlib=@<:@auto/yes/system/bundled/no@:>@],
+                             [MongoDB: Enable zlib for compression [default=auto]])],
+             [auto],
+             [no])
+
+PHP_ARG_WITH([mongodb-zstd],
+             [whether to enable zstd for compression],
+             [AS_HELP_STRING([--with-mongodb-zstd=@<:@auto/yes/no@:>@],
+                             [MongoDB: Enable zstd for compression [default=auto]])],
+             [auto],
+             [no])
+
 found_snappy="no"
 found_zlib="no"
 bundled_zlib="no"
 found_zstd="no"
 
-PKG_CHECK_MODULES([PHP_MONGODB_SNAPPY],[snappy],[
-  PHP_MONGODB_BUNDLED_CFLAGS="$PHP_MONGODB_BUNDLED_CFLAGS $PHP_MONGODB_SNAPPY_CFLAGS"
-  PHP_EVAL_LIBLINE([$PHP_MONGODB_SNAPPY_LIBS],[MONGODB_SHARED_LIBADD])
-  found_snappy="yes"
-],[
-  PHP_CHECK_LIBRARY([snappy],
-                    [snappy_uncompress],
-                    [have_snappy_lib="yes"],
-                    [have_snappy_lib="no"])
-
-  AC_CHECK_HEADER([snappy-c.h],
-                  [have_snappy_headers=yes],
-                  [have_snappy_headers=no])
-
-  if test "$have_snappy_lib" = "yes" -a "$have_snappy_headers" = "yes"; then
-    PHP_ADD_LIBRARY([snappy],,[MONGODB_SHARED_LIBADD])
+AS_IF([test "$PHP_MONGODB_SNAPPY" = "auto" -o "$PHP_MONGODB_SNAPPY" = "yes"],[
+  PKG_CHECK_MODULES([PHP_MONGODB_SNAPPY],[snappy],[
+    PHP_MONGODB_BUNDLED_CFLAGS="$PHP_MONGODB_BUNDLED_CFLAGS $PHP_MONGODB_SNAPPY_CFLAGS"
+    PHP_EVAL_LIBLINE([$PHP_MONGODB_SNAPPY_LIBS],[MONGODB_SHARED_LIBADD])
     found_snappy="yes"
+  ],[
+    PHP_CHECK_LIBRARY([snappy],
+                      [snappy_uncompress],
+                      [have_snappy_lib="yes"],
+                      [have_snappy_lib="no"])
+
+    AC_CHECK_HEADER([snappy-c.h],
+                    [have_snappy_headers=yes],
+                    [have_snappy_headers=no])
+
+    if test "$have_snappy_lib" = "yes" -a "$have_snappy_headers" = "yes"; then
+      PHP_ADD_LIBRARY([snappy],,[MONGODB_SHARED_LIBADD])
+      found_snappy="yes"
+    fi
+  ])
+
+  if test "$PHP_MONGODB_SNAPPY" = "yes" -a "$found_snappy" = "no"; then
```

Comment:
I have system libraries for all three compression libraries. Rather than remove them, I tested this by flipping this (and related conditionals) to `"$found_snappy" = "yes"` and checking for an error when the system library was picked up.

I also tested "system" and "bundled" interactions for zlib.

When system libraries are found, you'll see the following configure output:

```
checking for PHP_MONGODB_SNAPPY... yes
checking for PHP_MONGODB_ZLIB... yes
checking for PHP_MONGODB_ZSTD... yes

...

MONGODB_SHARED_LIBADD                            : -lsnappy -lz -lzstd
```

Using bundled zlib doesn't result in any configure output, so to test that you'll want to check the definitions in the generated `mongoc-config.h` output file (e.g. `MONGOC_ENABLE_COMPRESSION_X`).

Compiling the driver and checking `phpinfo()` output also works.